File: algo-pad-array/python/pad_array_spec.py
# Write unit tests!
from pad_array import pad

print(pad([1, 2, 3], 5, 7) == [1, 2, 3, 7, 7])
print(pad([1, 2, 3, 4], 5, 'hi') == [1, 2, 3, 4, 'hi'])
print(pad([1, 2, 3, 4, 5], 4, 18) == [1, 2, 3, 4, 5])
print(pad([1], 3, None) == [1, None, None])
print(pad([1], 1, None) == [1])
print(pad([1,2,3], 5) == [1,2,3,None,None])
print(pad([1,2,3], 5, 'apple') == [1,2,3,'apple', 'apple'])
print(pad([1,2,3], 2) == [1,2,3])
print(pad([1,2,3], -1) == [1,2,3])
print(pad([1,2,3], 6) == [1,2,3, None, None, None])
print(pad([1,2,3], 4, "coffee") == [1,2,3, "coffee"])

File: algo-pad-array/js/padArraySpec.js
// Write unit tests!
var pd = require("./padArray");

tests = [
    {'in1':[1,2,3], 'in2':2, 'in3':null,           'ans':[1,2,3]},
    {'in1':[1,2,3], 'in2':5, 'in3':null,           'ans':[1,2,3,null,null]},
    {'in1':[1,2,3], 'in2':5, 'in3':'apple',        'ans':[1,2,3,'apple','apple']},
    {'in1':[1,2,3], 'in2':0, 'in3':null,           'ans':[1,2,3]},
];

testedFunction = pd.pad;
count = 0;
total = tests.length;

for (let i = 0; i < total; i++) {
  let passed = true;
  testVal1 = testedFunction(tests[i]['in1'],tests[i]['in2'],tests[i]['in3']);
  testVal2 = tests[i]['ans'];
  if (testVal1.length != testVal2.length){
    passed = false;
  } else {
    for (let i = 0; i < testVal1.length; i++) {
        if (testVal1[i] !== testVal2[i]) {
            passed = false; 
        }
    }
  }
  if (passed) {
    console.log(`Passed test ${i+1} of ${total}`);
    count++;
  } else {
    console.log(`Failed test ${i+1} of ${total}`);
  }
}

console.log(`${count}/${total} tests passed.`);